internal/core/operations/healthcontroller/health_controller_executor.go

```diff
@@ -232,6 +237,15 @@ func (ex *SchedulerHealthControllerExecutor) findAvailableAndExpiredRooms(ctx co
 	return availableRoomsIDs, expiredRoomsIDs
 }
 
+func (ex *SchedulerHealthControllerExecutor) isGameRoomInstanceReady(gameRoomId string, instances []*game_room.Instance) bool {
```

Comment:
@caiordjesus this method will be called 1 time for each game room, what about building a **map[gameRoomId]*game_room.Instance** before `findAvailableAndExpiredRooms` function call and pass it as an argument?